internal/models/pubkey_model.go

```diff
@@ -25,42 +14,36 @@ type Pubkey struct {
 	// Public key body encoded in base64 (.pub format). Required.
 	Body string `db:"body" json:"body" validate:"required,sshPubkey"`
 
-	// Public key SHA256 fingerprint. Generated (read-only).
-	Fingerprint string `db:"fingerprint" json:"fingerprint"`
-}
-
-// FingerprintAWS calculates fingerprint used by AWS.
-// AWS calculates MD5 sums differently then all the other tools and use format DER as the base for the sub.
-// see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/verify-keys.html for more details.
-func (pk *Pubkey) FingerprintAWS() (string, error) {
-	pkey, _, _, _, err := ssh.ParseAuthorizedKey([]byte(pk.Body))
-	if err != nil {
-		return "", fmt.Errorf("failed to calculate fingerprint for %s: %w", pk.Name, err)
-	}
-	// when rsa key
-	if pkey.Type() == "ssh-rsa" {
-		return fingerprintRsaAWS(pkey)
-	}
-	// when ed25519 key
-	// this is the same as what we store, but better be sure here
-	return strings.TrimLeft(ssh.FingerprintSHA256(pkey), "SHA256:") + "=", nil
+	// Key type: "ssh-ed25519" or "ssh-rsa".
+	Type string `db:"type" json:"type" validate:"required"`
+
+	// SHA256 base64 encoded fingerprint with padding without any prefix. Note OpenSSH
+	// typically prints the fingerprint without padding: ssh-keygen -l -f $HOME/.ssh/key.pub
+	// The length is exactly 45 characters including the padding.
+	// Example: "gL/y6MvNmJ8jDXtsL/oMmK8jUuIefN39BBuvYw/Rndk="
+	Fingerprint string `db:"fingerprint" json:"fingerprint" validate:"len=45"`
+
+	// MD5 fingerprint stored as hexadecimal with colons without any prefix. To generate
+	// such fingerprint: ssh-keygen -l -E md5 -f $HOME/.ssh/key.pub
+	// Example: "89:c5:99:b5:33:48:1c:84:be:da:cb:97:45:b0:4a:ee"
+	FingerprintLegacy string `db:"fingerprint_legacy" json:"fingerprint_legacy" validate:"len=48"`
+
+	// MD5 fingerprint stored as hexadecimal with colons without any prefix from key in PEM format.
+	// This format is specific to AWS. To generate such fingerprint:
+	// ssh-keygen -e -f $HOME/.ssh/key.pub -m pkcs8 | openssl pkey -pubin -outform der | openssl md5 -c
+	// Example: "c4:ba:72:45:16:a9:2c:39:c3:99:8d:e7:16:01:9c:77"
+	FingerprintPEM string `db:"fingerprint_pem" json:"fingerprint_pem" validate:"len=48"`
 }
```

Comment:
This is the gist of the patch, three new columns:

* type
* legacy fingerprint (MD5 OpenSSH)
* PEM fingerprint (MD5 from PEM - speciality of AWS perhaps)

And a helper method that selects the proper fingerprint depending on type for AWS.